design/open/ttv-wizard-mvp.md

```diff
@@ -0,0 +1,330 @@
+
+### Wizard Templates
+
+The following templates will be saved within the lakeFS binary and be directed by the Wizard component to be expanded:
+* SparkEMR.conf.tt
+* SparkDatabricks.conf.tt
+* SparkXML.conf.tt
+* README.md.tt
+* MetastoreEMR.conf.tt
+* MetastoreDatabricks.conf.tt
+* MetastoreXML.conf.tt
+
+---
+
+## APIs
+
+### Templating Service
+
+- **Endpoint**: `/api/v1/templates`  
+- **Request**:
+  - Method: `GET`
+  - Parameters:
+    - Template URL (`template_location`): `string` - retrieved from query string.
+    - Any other configurations required for the templates: `string` - retrieved from query string
+- **Response**:
+  - Return value: The expanded template
+  - Headers:
+    - `Content-Type` - The template's content type.
+- **Errors**:
+  1. *403- Forbidden*: The requesting user is forbidden from accessing the configurations or functionality (like generating credentials).
+  2. *400- Bad request*: The request is missing information necessary for the template’s expansion.
+  3. *500- Internal server error*: The lakeFS server cannot access some provided template locations.
+
+---
+
+## Possible Flows
+
+### Example template
+
+**Databricks Spark configurations**  
+Name: *databricksConfig.props.tt*
+```properties
+spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem
+{{with $creds := new_credentials}}
+spark.hadoop.fs.s3a.access_key={{$creds.Key}}
+spark.hadoop.fs.s3a.secret_key={{$creds.Secret}}
+{{end}}
+spark.hadoop.fs.s3a.endpoint={{ .querystring.lakefs_url }}
+spark.hadoop.fs.s3a.path.style.access=true
+```
+
+**Local Metastore configurations**
+Name: *localMetastoreConfig.xml.tt*
+```xml
+<configuration>
+    <property>
+        <name>fs.s3a.path.style.access</name>
+        <value>true</value>
+    </property>
+    <property>
+        <name>fs.s3.impl</name>
+        <value>org.apache.hadoop.fs.s3a.S3AFileSystem</value>
+    </property>
+    <property>
+        <name>fs.s3a.endpoint</name>
+        <value>{{ .querystring.lakefs_url }}</value>
+    </property>
+    {{with $creds := new_credentials}}
+    <property>
+        <name>fs.s3a.access.key</name>
+        <value>{{$creds.Key}}</value>
+    </property>
+    <property>
+        <name>fs.s3a.secret.key</name>
+        <value>{{$creds.Secret}}</value>
+    </property>
+    {{end}}
+</configuration>
+```
+
+### Happy flow - All steps
+
+1. The user clicks on ‘Create Repository’, then clicks on ‘Spark Quickstart’
+2. The wizard starts by showing an input textbox for the users to type the repo name. The user types ‘spark-repo’
+3. The wizard creates a repo named ‘spark-repo’ and sets a default ‘main’ branch.
+4. The wizard asks the users if they want to import existing data to lakeFS. The user specifies the location of the bucket (after they validated that the lakeFS role has the right permissions and that the bucket has the correct policy), and clicks ‘OK’.
+   * An object counter will show the progress of the import process and will signal once it’s over.
```

Comment:
I would really like this to link to the import design.  Or, if we need to change import, let's call it out and remain with one *good* implementation of *the same* feature.

Here, if a counter makes sense during the wizard, it also makes sense during a plain import.

Reply:
We don't need to change the import. We don't use it as it is, just leverage it's functionality, not the exact behavior.
It's not the same feature.
There is a counter in the plain import.

Reply:
The UI-Import is an assembly of 5-6 different API calls. To get the behaviour you need here you'll need to replace 1-2 API calls, but it's all pretty straight forward.